File: clients/kotlin/src/main/kotlin/model/Vec2Double.kt
package model

import util.StreamUtil

class Vec2Double {
    var x: Double = 0.0
    var y: Double = 0.0
    constructor() {}
    constructor(x: Double, y: Double) {
        this.x = x
        this.y = y
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Vec2Double {
            val result = Vec2Double()
            result.x = StreamUtil.readDouble(stream)
            result.y = StreamUtil.readDouble(stream)
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeDouble(stream, x)
        StreamUtil.writeDouble(stream, y)
    }
}

File: clients/javascript/model/vec2double.js
class Vec2Double {
    constructor (x, y) {
        this.x = x;
        this.y = y;
    }

    static async readFrom (stream) {
        const x = await stream.readDouble();
        const y = await stream.readDouble();
        return new Vec2Double(x, y);
    }

    async writeTo (stream) {
        await stream.writeDouble(this.x);
        await stream.writeDouble(this.y);
    }
    
    toString () {
        return 'Vec2Double(' +
            this.x + ',' +
            this.y +
            ')';
    }
}

module.exports.Vec2Double = Vec2Double;
